libvast/src/transform_steps/select.cpp

```diff
@@ -0,0 +1,75 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/transform_steps/select.hpp"
+
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/error.hpp"
+#include "vast/expression.hpp"
+#include "vast/logger.hpp"
+#include "vast/plugin.hpp"
+#include "vast/table_slice_builder_factory.hpp"
+
+#include <arrow/type.h>
+#include <caf/expected.hpp>
+
+namespace vast {
+
+select_step::select_step(std::string expr)
+  : expression_(caf::none) { // FIXME: Is caf::none OK?
+  auto e = to<vast::expression>(std::move(expr));
+  if (!e) {
+    expression_ = e;
+    return;
+  }
+  // FIXME: Is normalize useful here?
+  expression_ = normalize_and_validate(*e);
```

Comment:
It probably makes sense to print a `VAST_ERROR()` here already if this returns an error, so this is immediately visible on startup. (also long-term we need to think about a better way to report errors, probably a separate `setup()` function)